runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -0,0 +1,421 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.twitch;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import javax.inject.Inject;
+import net.engio.mbassy.listener.Handler;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.GameState;
+import net.runelite.api.Varbits;
+import net.runelite.api.events.CommandExecuted;
+import net.runelite.api.events.ConfigChanged;
+import static net.runelite.client.callback.Hooks.log;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import org.apache.commons.lang3.StringUtils;
+import org.kitteh.irc.client.library.Client;
+import org.kitteh.irc.client.library.event.channel.ChannelCtcpEvent;
+import org.kitteh.irc.client.library.event.channel.ChannelMessageEvent;
+import org.kitteh.irc.client.library.feature.twitch.TwitchDelaySender;
+import org.kitteh.irc.client.library.feature.twitch.TwitchListener;
+import org.kitteh.irc.client.library.feature.twitch.event.RoomStateEvent;
+import org.kitteh.irc.client.library.feature.twitch.event.UserNoticeEvent;
+import org.kitteh.irc.client.library.feature.twitch.event.UserStateEvent;
+
+@PluginDescriptor(
+	name = "Twitch",
+	enabledByDefault = false
+)
+public class TwitchPlugin extends Plugin
+{
+	@Inject
+	private net.runelite.api.Client client;
+
+	@Inject
+	private TwitchConfig config;
+
+	private Client irc;
+	private boolean subscribed;
+	private boolean mod;
+	private boolean broadcaster;
+	private boolean subsOnly;
+	private boolean emoteOnly;
+	private String badges = "";
+
+	@Provides
+	TwitchConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(TwitchConfig.class);
+	}
+
+	@Override
+	protected void startUp()
+	{
+		twitchShutdown(irc);
+		irc = twitchSetup(config.oauthToken(), config.channelName(), config.nickName());
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		twitchShutdown(irc);
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (!event.getGroup().equals("twitch"))
+		{
+			return;
+		}
+
+		if (event.getKey().equals("channel") || event.getKey().equals("oauth") || event.getKey().equals("nick"))
```

Comment:
I believe any time you'd ever change your nickname, you're changing the twitch account you're using to chat, which would require a reauth and therefore a reconnection.

Reply:
yeah sorry this comment got lost on my UI when I hit back by accident, i'd have duplicated it but I didn't re-add it since I realised the twitch use-case is accounnt based.